src/Features/Customer/UserProfile/Profile.js

```diff
@@ -1,162 +1,148 @@
 import React, { useState } from 'react';
 
+import { useMutation } from 'react-query';
+import { useSelector, useDispatch } from 'react-redux';
+
 import CommonGridBasedForm from '../../../components/CommonGridBasedForm';
 import { TEXT_FIELD } from '../../../components/CommonGridBasedForm/FieldTypes';
 import { emailRegex } from '../../../redux/ActionTypes';
 import { contactRegex } from '../../../scripts/constants';
+import { fieldChangeHandler, validateOnSubmit } from '../../../util/CommonGridBasedFormUtils';
+import { updateUserData } from '../../Auth/actions';
+import { editUser } from './mutation';
 
 const AddUser = () => {
-  const [onSaveSuccess, setOnSaveSuccess] = useState(false);
-  const [isEdit] = useState(true);
-  const validateOnSubmit = () => {
-    let isValid = true;
-
-    const ValidateArray = fields.map((field) => {
-      if (
-        field.value === '' ||
-        field.value === undefined ||
-        field.value === null ||
-        (field.value.constructor.name == 'Array' && field.value.length === 0)
-      ) {
-        isValid = false;
-
-        field.errorMessage = field.label + ' field cannot be empty';
-
-        field.isValid = false;
-
-        return field;
-      }
-
-      field.isValid = true;
-
-      field.errorMessage = '';
-
-      !isValid ? null : (isValid = field.isValid);
-
-      return field;
-    });
-
-    setFields(ValidateArray);
-
-    return isValid;
-  };
-
-  const [, setEmail] = useState('');
-
-  const [, setContact] = useState('');
-
-  const [, setName] = useState('');
+  const { id, email, name, token } = useSelector((state) => {
```

Comment:
Is there was a special purpose to put all the user related stuff in redux and if it is then using redux observable is a good choice instead of react query

Reply:
This was done because when users logs in, the response contains its data, which gets saved in redux. Now, when the user goes to profile, the fields get autofill by this data which is instantaneous and does not need any loading whatsoever. I think, Observable will bring complexity and will be overkill for this. One alternative, we can do is fetch the data from the server.

Reply:
Point is of something is being stored in user profile why do we need to use the query from component and update the data in store again and again ??

Reply:
If we won't update the data in store after edit, it will bring the old data on page refresh or remount.

Reply:
This is what this component does.

1. Brings data from store and fill it on fields.
2. User edits the data, the data gets updated on server, if the successfully updated, then update the local data in store as well.